base/compiler/ssair/inlining.jl

```diff
@@ -1300,6 +1300,9 @@ function assemble_inline_todo!(ir::IRCode, state::InliningState)
         end
 
         if isa(info, OpaqueClosureCallInfo)
+            # ignore the `@noinline` annotation for opaque closure call as we force the constant-prop'
+            # for this opaque closure call in `const_prop_methodinstance_heuristic` ?
+            # flag &= ~IR_FLAG_NOINLINE
```

Comment:
@Keno do you think there is any circumstance where we want constant-prop' for a opaque closure call while w  want it not to be inlined?

Reply:
This sounds conceptually broken
```suggestion
```